samples/ironside_se/protectedmem_periphconf/src/main.c

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA.
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/sys/printk.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/device.h>
+#include <zephyr/sys/util.h>
+#include <string.h>
+#include <zephyr/arch/cpu.h>
+
+/* Get periphconf_partition address and size from device tree */
+#define PERIPHCONF_PARTITION_ADDRESS DT_FIXED_PARTITION_ADDR(DT_NODELABEL(periphconf_partition))
+#define PERIPHCONF_PARTITION_SIZE DT_REG_SIZE(DT_NODELABEL(periphconf_partition))
+
+/* Write to the last 4 bytes in a 16-byte aligned area within periphconf */
+/* Find the last 16-byte aligned area within the partition */
+#define MRAM_16BYTE_ALIGN 16
+#define PARTITION_END (PERIPHCONF_PARTITION_ADDRESS + PERIPHCONF_PARTITION_SIZE)
+/* Find the start of the last 16-byte aligned area (round down to 16-byte boundary) */
+#define LAST_16BYTE_AREA_START ((PARTITION_END - MRAM_16BYTE_ALIGN) & ~(MRAM_16BYTE_ALIGN - 1))
+/* Write to the last 4 bytes of that 16-byte aligned area */
+#define TEST_WRITE_ADDRESS (LAST_16BYTE_AREA_START + MRAM_16BYTE_ALIGN - sizeof(uint32_t))
```

Comment:
Are any of these comments needed? IMO the macro names are self-explanatory.

Reply:
fixed